client/mm/mm.go

```diff
@@ -1029,7 +1161,8 @@ func (m *MarketMaker) Run(ctx context.Context, pw []byte, alternateConfigPath *s
 		}
 	}()
 
-	m.ctx, m.die = context.WithCancel(ctx)
+	ctx, die := context.WithCancel(m.ctx)
```

Comment:
This new `ctx` needs to be passed into the bots instead of `m.ctx`. Calling `die` doesn't stop the bots right now.